src/Raven.Server/Documents/Handlers/BulkInsertHandler.cs

```diff
@@ -69,7 +66,23 @@ public unsafe void InsertDocuments()
                         BulkBufferInfo current;
                         try
                         {
-                            current = _fullBuffers.Take(Database.DatabaseShutdown);
+                            int timeToWait = 500;
+                            int retry = 60;
+                            while (_fullBuffers.TryTake(out current, timeToWait, Database.DatabaseShutdown) == false)
+                            {
+                                if (--retry == 0)
+                                {
+                                    throw new InvalidOperationException("Server waited " + (timeToWait * retry) / 1000 + " Seconds, but the client didn't send any documents or completion message");
+                                }
+                                try
+                                {
+                                    _webSocket.SendAsync(WaitingMessage, WebSocketMessageType.Text, true, Database.DatabaseShutdown).Wait();
```

Comment:
Note that we CANNOT do this.
You are sending data on the web socket from two different threads.